src/pybind/mgr/dashboard/services/ganesha.py

```diff
@@ -491,6 +492,8 @@ def fill_keys(self):
 
     def create_path(self, path):
         cfs = CephFS(self.fs_name)
+        if path == os.sep:
+            return
```

Comment:
Why not change it here?
https://github.com/ceph/ceph/blob/6607cd853869e6b2b806e053cd973d69e6e44e42/src/pybind/mgr/dashboard/services/cephfs.py#L141-L147

Reply:
I don't want to touch the `mk_dirs` implementation in the cephfs service because it's also used by CephFS components in the Dashboard. This PR is a actually a partial revert of #30256

Reply:
@varshar16 is the PR good to go from your POV?

Reply:
It would be nice to have this check in `mk_dirs` definition. Instead of doing checks outside the method call. I see in cephfs controller `mk_dirs` is only used here https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/cephfs.py#L393-L400.

Reply:
Let's guard the path outside the implementation ATM. I did a refactor today but I found the API needs a refactor which might modify the implementation:

- Make `path` in `POST /api/cephfs/<fs_id>/tree` and `DELETE /api/cephfs/<fs_id>/tree` behavior consistent
- `/` should be treated as an existing path.
- Like `mkdir` and `rmdir` commands, if the path exists then an error should be raised. Unless a `force` parameter is passed (e.g. `rm -f` and `mkdir -p`).

I'll capture these into an issue.